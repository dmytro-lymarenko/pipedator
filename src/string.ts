import { createValidator } from './core';

export function string() {
	return createValidator({
		message: 'Value should be a string',
		validate: (value, ctx) => (typeof value === 'string' ? null : { value, message: ctx.message, path: ctx.path }),
	});
}

export function empty() {
	return createValidator({
		message: 'Value should be empty',
		validate: (value, ctx) => (value.length === 0 ? null : { value, message: ctx.message, path: ctx.path }),
	});
}
