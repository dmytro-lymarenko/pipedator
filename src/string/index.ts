import { test } from '../test';

export function string<ValidValue = any>(message?: string) {
	return test<ValidValue>(value => typeof value === 'string', message || 'Value should be a string');
}
