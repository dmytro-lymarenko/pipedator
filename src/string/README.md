- **Description**

Creates a validator to check if a value is a string or not.

- **How to import**

```typescript
import { string } from 'pipedator';
// or
import { string } from 'pipedator/lib/string';

```
- **Signature**

```typescript
function string<ValidValue = any>(message?: string): Validator<ValidValue>;
```
- **Parameters**

  - `message` - (optional) custom message.


- **Usage**

```typescript
string().validate('Some text'); // valid
string().validate(5); // invalid
string('Custom message').validate(null); // invalid with 'Custom message'
```