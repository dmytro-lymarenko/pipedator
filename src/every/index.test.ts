import { every } from './index';
import { string } from '../string';

describe('every()', () => {
	it('should validate an empty array', () => {
		expect(every(string()).validate([])).toBeNull();
	});

	it('should validate an array when all items are string', () => {
		expect(every(string()).validate(['first', 'last'])).toBeNull();
	});

	it('should fail when at least one item is not a string', () => {
		expect(every(string()).validate(['first', 1, 'last'])).toMatchSnapshot();
	});

	it('should support custom message', () => {
		expect(every(string(), 'Custom message').validate([1, 'test'])).toMatchSnapshot();
	});

	it('should throw an error when a value is not an array', () => {
		expect(every(string()).validate(1)).toMatchSnapshot();
	});
});
