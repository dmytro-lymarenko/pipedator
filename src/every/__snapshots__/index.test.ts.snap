// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`every() should fail when at least one item is not a string 1`] = `
Object {
  "errors": Array [
    Object {
      "errors": Array [],
      "message": "Value should be a string",
      "path": Array [
        "1",
      ],
      "value": 1,
    },
  ],
  "message": "Every value should follow the rule: Value should be a string",
  "path": Array [],
  "value": Array [
    "first",
    1,
    "last",
  ],
}
`;

exports[`every() should support custom message 1`] = `
Object {
  "errors": Array [
    Object {
      "errors": Array [],
      "message": "Value should be a string",
      "path": Array [
        "0",
      ],
      "value": 1,
    },
  ],
  "message": "Custom message",
  "path": Array [],
  "value": Array [
    1,
    "test",
  ],
}
`;
